### PR TITLE
Add fixAccounts script

### PR DESCRIPTION
**Pull request by gregorylegarec**
~~Waiting for https://github.com/cozy/ACH/pull/19~~

This PR adds a fixAccounts script which:

* Fix legacy accounts with `dir_id` attribute
* Fix legacy accounts with `folderId` attribute
* Move `folderPath` attribute to `path.folderPath`
* Ensure consistency between `auth.folderPath` and `auth.namePath`
* Retrive destination folder information from related trigger.

### CODE REVIEW

File: scripts/fixAccounts.js [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+    selector: {
+      message: {
+        account: {
+          $eq: accountId
+        }
+      }
+    }
+  })
+  return results[0]
+}
```
[the remark this answers is not in the document]
Comment (gregorylegarec) [reply scripts/fixAccounts.js]: Exactly my thought.
When we will work on Collect Applications, I will get all this logic here into a dedicated module in `libs/doctypes`. The main idead being able to repair/update accounts.

**Comment by gregorylegarec** [comment]
Thanks for the exhaustive review @ptbrowne !

I agree with you about splitting the main logic. I started to do so but it was implying passing the information about `needUpdate` over all tests, or update in every test.

If I split the logic, I think I will make an update in every function. What do you think ?

File: scripts/fixAccounts.spec.js [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+  const mockedTrigger = {
+    message: {
+      folder_to_save: 'c018ad350813c21f85137726a008224f'
+    }
+  }
+
+  beforeEach(() => {
+    ;['log', 'info'].forEach(logger => {
+      capturedLoggers[logger] = console[logger]
+      console[logger] = jest.fn()
```
[the remark this answers is not in the document]
Comment (gregorylegarec) [reply scripts/fixAccounts.spec.js]: Actually this does not prevent the console logs to appear during tests, that was what I wished to do.

[the remark this answers is not in the document]
Comment (ptbrowne) [reply scripts/fixAccounts.spec.js]:
```
jest.spyOn(console, logger).mockImplementation(() => {})
```

File: scripts/fixAccounts.spec.js [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+    const unexpectedAccount = {
+      ...expectedAccount,
+      folderPath: '/Administrative/Should/Be/Ignored'
+    }
+    await fixAccount(client, unexpectedAccount, false)
+    expect(client.data.update.mock.calls.length).toBe(1)
+    expect(client.data.update.mock.calls[0][2]).toMatchObject(expectedAccount)
+  })
+
+  it('makes `auth.folderPath` and `auth.namePath` consistent', async () => {
```
[the remark this answers is not in the document]
Comment (gregorylegarec) [reply scripts/fixAccounts.spec.js]: *coherent

File: scripts/fixAccounts.spec.js [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+      auth: {
+        ...expectedAccount.auth,
+        folderPath: '/Administrative/Test/',
+        namePath: 'claude_cozycloud_cc'
+      }
+    }
+    await fixAccount(client, unexpectedAccount, false)
+    expect(client.data.update.mock.calls.length).toBe(1)
+    expect(client.data.update.mock.calls[0][2]).toMatchObject(expectedAccount)
+  })
```
[the remark this answers is not in the document]
Comment (gregorylegarec) [reply scripts/fixAccounts.spec.js]: Almost, there is a '/' at the end of the `folderPath` in the second one.

**Pull request merged**